include/lsst/afw/detection/Footprint.h

```diff
@@ -291,7 +192,7 @@ class Footprint : public lsst::daf::base::Citizen,
      *
      */
     template <typename MaskPixelT>
-    void intersectMask(
+    std::shared_ptr<Footprint> intersectMask(
         image::Mask<MaskPixelT> const & mask,
         MaskPixelT bitmask=~0x0
     );
```

Comment:
I would very much like to get rid of this method.  I think I added it (or Martin, a programmer I worked with as a graduate student, did), and it's always had confusing behavior given its name.  And I think it can be replaced by the more intuitive `SpanSet-Mask` operations we discussed a few days ago.  Do you know where it's being used?